compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1134,6 +1160,106 @@ public void visit(BLangSimpleVariable varNode) {
         env.enclPkg.isListenerAvailable |= Symbols.isFlagOn(varNode.symbol.flags, Flags.LISTENER);
     }
 
+    @Override
+    public void visit(BLangUserDefinedType userDefinedType) {
+        if (Types.getReferredType(userDefinedType.getBType()).getKind() == TypeKind.RECORD) {
+            visitTypedesc(userDefinedType.pos, userDefinedType.getBType(), Collections.emptyList());
+        }
+    }
+
+    @Override
+    public void visit(BLangValueType valueType) {
+    }
+
+    @Override
+    public void visit(BLangUnionTypeNode unionTypeNode) {
+        unionTypeNode.memberTypeNodes.forEach(typeNode -> typeNode.accept(this));
+    }
+
+    @Override
+    public void visit(BLangRecordTypeNode recordTypeNode) {
+        for (BLangSimpleVariable field : recordTypeNode.fields) {
+            if (field.typeNode != null) {
+                field.typeNode.accept(this);
+            }
+        }
+        if (recordTypeNode.restFieldType != null) {
+            recordTypeNode.restFieldType.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangArrayType arrayType) {
+        arrayType.elemtype.accept(this);
+    }
+
+    @Override
+    public void visit(BLangConstrainedType constrainedType) {
+        constrainedType.constraint.accept(this);
+    }
+
+    @Override
+    public void visit(BLangErrorType errorType) {
+        if (errorType.detailType != null) {
+            errorType.detailType.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangFunctionTypeNode functionTypeNode) {
+    }
+
+    @Override
+    public void visit(BLangBuiltInRefTypeNode builtInRefTypeNode) {
+    }
+
+    @Override
+    public void visit(BLangTableTypeNode tableTypeNode) {
+        tableTypeNode.constraint.accept(this);
+        if (tableTypeNode.tableKeyTypeConstraint != null) {
+            tableTypeNode.tableKeyTypeConstraint.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangTupleTypeNode tupleTypeNode) {
+        tupleTypeNode.memberTypeNodes.forEach(member -> member.accept(this));
+        if (tupleTypeNode.restParamType != null) {
+            tupleTypeNode.restParamType.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangStreamType streamType) {
+        streamType.type.accept(this);
+        streamType.constraint.accept(this);
+        if (streamType.error != null) {
+            streamType.error.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangTableKeyTypeConstraint keyTypeConstraint) {
+        keyTypeConstraint.keyType.accept(this);
+    }
+
+    @Override
+    public void visit(BLangObjectTypeNode objectTypeNode) {
+        for (BLangSimpleVariable field : objectTypeNode.fields) {
+            if (field.typeNode != null) {
+                field.typeNode.accept(this);
+            }
+        }
```

Comment:
How about the param/return types in methods?

Reply:
Removed this logic.